test/com/arpnetworking/metrics/portal/TestUtil.java

```diff
@@ -0,0 +1,85 @@
+
+    /**
+     * Generates the JDBC url with a unique port number and database name.
+     *
+     * @return The JDBC url.
+     */
+    public static String generateJdbcUrl() {
+        final int nextCounter = UNIQUE_COUNTER.getAndIncrement();
+        final int port = BASE_PORT + nextCounter;
+        return "jdbc:h2:./target/h2/metrics" +
+                nextCounter +
+                ":portal;AUTO_SERVER=TRUE;AUTO_SERVER_PORT=" +
+                port +
+                ";MODE=PostgreSQL;INIT=create schema if not exists portal;DB_CLOSE_DELAY=-1";
+    }
+
+    public static boolean isAlertEbeanEquivalent(final Alert alert, final models.ebean.Alert ebeanAlert) {
+        return Objects.equals(alert.getId(), ebeanAlert.getUuid())
+                && Objects.equals(alert.getCluster(), ebeanAlert.getCluster())
+                && Objects.equals(alert.getMetric(), ebeanAlert.getMetric())
+                && isNagiosExtensionEbeanEquivalent(alert.getNagiosExtension(), ebeanAlert.getNagiosExtension())
+                && Objects.equals(alert.getService(), ebeanAlert.getService())
+                && Objects.equals(alert.getName(), ebeanAlert.getName())
+                && Objects.equals(alert.getOperator(), ebeanAlert.getOperator())
+                && Objects.equals(alert.getPeriod(), Period.seconds(ebeanAlert.getPeriod()))
+                && Objects.equals(alert.getStatistic(), ebeanAlert.getStatistic())
+                && Objects.equals(alert.getValue().getUnit(), Optional.of(ebeanAlert.getQuantityUnit()))
+                && alert.getValue().getValue() == ebeanAlert.getQuantityValue()
+                && Objects.equals(alert.getContext(), ebeanAlert.getContext());
+    }
+
+    public static boolean isNagiosExtensionEbeanEquivalent(
+            final NagiosExtension extension,
+            final models.ebean.NagiosExtension ebeanExtension) {
+        return Objects.equals(extension.getSeverity(), ebeanExtension.getSeverity())
+                && Objects.equals(extension.getNotify(), ebeanExtension.getNotify())
+                && extension.getMaxCheckAttempts() == ebeanExtension.getMaxCheckAttempts()
+                && extension.getFreshnessThresholdInSeconds() == ebeanExtension.getFreshnessThreshold();
+    }
+
+    public static boolean isExpressionEbeanEquivalent(
+            final Expression expression,
+            final models.ebean.Expression ebeanExpression) {
+        return Objects.equals(expression.getId(), ebeanExpression.getUuid())
+                && Objects.equals(expression.getCluster(), ebeanExpression.getCluster())
+                && Objects.equals(expression.getMetric(), ebeanExpression.getMetric())
+                && Objects.equals(expression.getScript(), ebeanExpression.getScript())
+                && Objects.equals(expression.getService(), ebeanExpression.getService());
+    }
+
+    private static final AtomicInteger UNIQUE_COUNTER = new AtomicInteger(1);
+    private static final int BASE_PORT = 50000;
+}
```

Comment:
So it looks like the only method re-used across tests in here is the jdbc connection. So you just need an H2ConnectionStringFactory (or some such).